lib/aiko/led.py

```diff
@@ -71,8 +72,35 @@ def apply_dim(color, dimmer=None):
 # TODO: Now fails when called by exec(), see lib/aiko/mqtt.py
 # return tuple([int(element * dimmer) for element in color])
 
-def fill(color):
+# Allow setting dim from code or MQTT
+def set_dim(dimmer):
+    global dim
+    dim = dimmer
+
+# this is used to turn the neopixels back on to default brightness
+def reset_dim():
+  parameter = configuration.main.parameter
+  set_dim(parameter("dim", configuration.led.settings))
+
+# Take from -0.9 to 0.9 (+-0.1 is more typical) and adjust diming
+# value. Make sure it stays within 0 to 1
+def change_dim(change):
+    global dim
+    dim += change
+    if dim<0: dim=0
+    if dim>1: dim=1
```

Comment:
Code golf time!
```suggestion
def change_dim(change):
    global dim
    dim = max(min(dim + change, 1), 0)
```
… though if constraining it is that important maybe we should `dim = max(min(value, 1), 0)` in `set_dim` above? That'll also protect against out-of-range values coming over MQTT as an `led:dim` command.

Reply:
That's a reasonable suggestion indeed. Honestly, I don't have Andy's vision on how he likes interfaces, so let me see what he'd like in the end.